### PR TITLE
API adjustments and overall fine-tuning of -- not started/in-progress/completed Migrations

### LINKED ISSUE #243: Display an empty state if there are no plans in the state selected in the dropdown
If the user selects a state (Not Started, In Progress, Complete) in the dropdown for which there are currently no plans we need to display an empty state. In discussing with @priley86 this is much easier an clearer than my original hair brained idea of disabling and enabling states in the dropdown.

Mockups below:
migration plans empty state - not started
migration plans empty state - not started empty- in progress
migration plans empty state - not started empty - complete

### PR DESCRIPTION
**Pull request by AparnaKarve**
API adjustments were done to retrieve `miq_requests` and `miq_request_tasks` using 2 API calls.

Also a couple of other adjustments based on the findings from the real API mode testing.

Fixes https://github.com/priley86/miq_v2v_ui_plugin/issues/254 

Also fixes #243

### CODE REVIEW

**Comment by priley86** [comment]
testing this now...

**Comment by priley86** [comment]
I've noted the following after testing our UI integrated e2e this morning w/ @AparnaKarve  : 
* We are able to create plans and start them locally now using `simulate_queue_worker` in rails console

* Plan import/csv is working as expected. I can discover VMs or Import them and then select them for a plan. Once a VM is selected for one plan, it is no longer available for future plans. I noted a minor design issue for CSV highlight but I think we can follow up on this in future PR. It seems the icon is hard to see w/ the row hovered - @vconzola @serenamarie125 do you agree?
screen shot 2018-04-30 at 11 36 08 am

* We are not correctly displaying names of the plan. Previously we expected this to be the `plan.name` field however it should now be using `plan.description` field coming back in the API data it seems. 

* The logic needs to be updated to correctly display errors if a single task fails. The plan I ran was in "Error" status, however the UI logic shows a green check mark and completed successfully. It appears this is just an API mapping misinterpreted.

* The logic needs to be updated to show tasks that completed as failed as no longer running. This appears to still be showing a completed (failed) plan w/ the task in progress.

Some more screenshots:
screen shot 2018-04-30 at 12 58 55 pm
screen shot 2018-04-30 at 12 59 04 pm
screen shot 2018-04-30 at 12 59 09 pm


Thanks @AparnaKarve for starting this API integration 👍  - it appears we are much closer now. Let me know when you'd like me to test again...

File: app/javascript/react/screens/App/Overview/OverviewActions.js [hunk 1]
```diff
@@ -129,11 +129,11 @@ const _getTransformationPlansActionCreator = url => dispatch =>
     if (mockMode) {
       dispatch({
         type: `${FETCH_V2V_TRANSFORMATION_PLANS}_FULFILLED`,
-        payload: requestTransformationPlansData.response
+        payload: Promise.resolve(requestTransformationPlansData.response)
```
[the remark this answers is not in the document]
Comment (AparnaKarve) [reply app/javascript/react/screens/App/Overview/OverviewActions.js]: Ok thanks @michaelkro  - will change that

**Comment by priley86** [comment]
would you mind also cherry picking this commit since we are planning to merge this PR before the nightly:
https://github.com/priley86/miq_v2v_ui_plugin/commit/eca24e4d9c115e889195084a4c66f84539d67b1d

this contains the new empty state logic (from uxd-summit-booth branch).

**Comment by priley86** [comment]
ok, merging this in for nightly build and to avoid making this PR growing much more. This is a good first step towards finishing up API integration w/ the new UIs. We can iterate in future PRs. thanks @AparnaKarve ! 👏 👏 👏

**Comment by AparnaKarve** [comment]
@priley86 Thanks for all the testing, and some of those reducer tips!

**Pull request merged**